src/test/pthreads/test_pthreads.rs

```diff
@@ -0,0 +1,489 @@
+/*
+ * The Shadow Simulator
+ * See LICENSE for licensing information
+ */
+
+use std::error::Error;
+
+use test_utils::set;
+use test_utils::TestEnvironment as TestEnv;
+
+const NUM_THREADS: usize = 5;
+
+const SUCCESS: isize = 0;
+const MUTEX_LOCK_FAILED: isize = -1;
+const MUTEX_UNLOCK_FAILED: isize = -2;
+const COND_WAIT_FAILED: isize = -3;
+const COND_BROADCAST_FAILED: isize = -4;
+const NULL_THREAD_ARG: isize = -5;
+
+struct MuxSum {
+    /// Raw mutable pointer to pthread_mutex_t
+    pub mux: *mut libc::pthread_mutex_t,
+    /// Raw mutable pointer to u32
+    pub sum: *mut u32,
+}
+
+impl MuxSum {
+    /// Create a new MuxSum with allocated pthread_mux_t and *mut u32 vars
+    pub fn new() -> Result<Self, String> {
+        let sum = alloc_type::<u32>()?;
+        unsafe { *sum = 0 };
+        Ok(Self {
+            mux: init_mutex()?,
+            sum: sum,
+        })
+    }
+}
+
+impl Drop for MuxSum {
+    fn drop(&mut self) {
+        destroy_mutex(self.mux);
+        free_type(self.sum);
+    }
+}
+
+struct MuxTry {
+    /// Raw mutable pointer to pthread_mutex_t
+    pub mux1: *mut libc::pthread_mutex_t,
+    /// Raw mutable pointer to pthread_mutex_t
+    pub mux2: *mut libc::pthread_mutex_t,
+    /// Raw mutable pointer to pthread_cond_t
+    pub cond: *mut libc::pthread_cond_t,
+    pub num_locked: u32,
+    pub num_no_locked: u32,
+}
+
+impl MuxTry {
+    /// Create a new MuxTry with allocated pthread_mux_t and pthread_cond_t vars
+    ///
+    /// Returns error message if allocation fails
+    pub fn new() -> Result<Self, String> {
+        Ok(Self {
+            mux1: init_mutex()?,
+            mux2: init_mutex()?,
+            cond: init_cond()?,
+            num_locked: 0,
+            num_no_locked: 0,
```

Comment:
I think this should change from `num_no_locked` to `num_not_locked`. The original C name `numnolocked` doesn't make sense to me, and I think `num_not_locked` is more accurate.